docs/authoring.md

```diff
@@ -244,3 +244,4 @@ There are some template parameters that have a special meaning:
 |Name|Description|
 |---|---|
 |`server`| It is used to let the template know which server you want to use. In some cases, this may be required. For instance, when generating code that connects to a specific server. If your template need your users to specify a server, use this parameter.
+|`asyncapiFileDir`| It is used by `@asyncapi/generator-hooks#`[`createAsyncapiFile`](https://github.com/asyncapi/generator-hooks/blob/62c50cac6156110e1325152cb2599d916a310055/src/index.js#L5) to specify directory where original AsyncAPI file will be stored. Defining a default parameter value allows template maintainers to pre-set this location. |
```

Comment:
@Tenischev you went too far with docs here, `asyncapiFileDir` is not a special parameter, it is not part of generator code, it is in generic hooks but not mandatory and if someone what's his own hook that supports parameter with the same name, this is all fine

Reply:
I would say, it depends on how you consider this sentence 
> There are some template parameters that have a special meaning

Because this is only one sentence that describe these parameters, no information that "it must be a part of generator code" or "each template must mandatory have it" e.t.c.

From my point of view, the phrase "It is used by `@asyncapi/generator-hooks#createAsyncapiFile`" clearly explain by whom used this parameter.

Of course, i could move this explanation to the description `asyncapiFileDir.description`, but then it will not be a good `example` of `.tp-config.json` file, from my point of view

Reply:
That is a philosophical dispute 😄 
![](https://media.giphy.com/media/l2JdVEGgjUbCwovy8/giphy.gif)

the example will be just fine if you call it `dummyParameter`, it doesn't have to be a super real param example. `asyncapiFileDir` is simply not special at all, it is just a random parameter anyone can define without using generator-hooks library, while `server` is super special one https://github.com/asyncapi/generator/blob/master/lib/generator.js#L685

Reply:
removed